app/src/org/commcare/CommCareHeartbeatManager.java

```diff
@@ -0,0 +1,309 @@
+package org.commcare;
+
+import android.app.Activity;
+import android.content.Intent;
+import android.os.Handler;
+import android.os.Looper;
+import android.util.Base64;
+
+import org.commcare.activities.PromptUpdateActivity;
+import org.commcare.core.interfaces.HttpResponseProcessor;
+import org.commcare.core.network.ModernHttpRequester;
+import org.commcare.logging.AndroidLogger;
+import org.commcare.preferences.CommCareServerPreferences;
+import org.commcare.utils.SessionUnavailableException;
+import org.commcare.utils.StorageUtils;
+import org.commcare.utils.SyncDetailCalculations;
+import org.javarosa.core.io.StreamsUtil;
+import org.javarosa.core.model.User;
+import org.javarosa.core.services.Logger;
+import org.javarosa.core.util.externalizable.ExtUtil;
+import org.json.JSONException;
+import org.json.JSONObject;
+
+import java.io.ByteArrayInputStream;
+import java.io.DataInputStream;
+import java.io.IOException;
+import java.io.InputStream;
+import java.net.MalformedURLException;
+import java.net.URL;
+import java.util.Date;
+import java.util.HashMap;
+import java.util.Timer;
+import java.util.TimerTask;
+
+/**
+ * While active, this class is responsible for using a TimerTask to periodically ping the server
+ * with a "heartbeat" request, and then handle the response. The lifecycle of the TimerTask is
+ * tied to that of the CommCareSessionService; it should be started whenever a session service is
+ * started, and ended whenever a session service is ended for any reason.
+ *
+ * Currently, the primary content of the server's response to the heartbeat request will be
+ * information about potential binary or app updates that the app should prompt users to conduct.
+ *
+ * Created by amstone326 on 4/13/17.
+ */
+public class CommCareHeartbeatManager {
+
+    private static final long ONE_DAY_IN_MS = 24 * 60 * 60 * 1000;
+
+    private static final String TEST_RESPONSE =
```

Comment:
Where else would it go? It's a test response for the request that this class makes? (I'm also going to delete it once I'm able to test making the actual request and it works)